app/scripts/modules/openstack/common/selectField.directive.js

```diff
@@ -48,8 +48,12 @@ module.exports = angular.module('spinnaker.openstack.selectField.directive', [
 
             //if the selection changed as a result of updating the options
             if (scope.model !== previousSelection) {
-              //ensure that the change handler is invoked
-              scope.onChange();
+              //ensure that the change handler is invoked... but give time for the two-way binding to propagate the value
+              if( scope.onChange ){
+                $timeout(function() {
+                  scope.onChange();
+                }, 0);
+              }
```

Comment:
You can simplify #L52-56 to:
```javascript
$timeout(scope.onChange);
```
Even if there's no `onChange` set on the scope, it won't throw an exception.